yt/frontends/arepo/data_structures.py

```diff
@@ -14,20 +15,27 @@ def __init__(
         self,
         filename,
         dataset_type="arepo_hdf5",
-        unit_base=None,
+        units_override=None,
         smoothing_factor=2.0,
         index_order=None,
         index_filename=None,
         kernel_name=None,
         bounding_box=None,
-        units_override=None,
         unit_system="cgs",
         default_species_fields=None,
+        *,
+        unit_base=None,
     ):
+        if unit_base is not None:
+            issue_deprecation_warning(
+                "The unit_base argument is a deprecated alias to units_override"
+                "To define custom units, please use the units_override argument.",
+                since="4.1.0",
+            )
+            units_override = unit_base
```

Comment:
Initially I didn't want to change the behaviour: this is essentially a fix for a feature that I'm immediately deprecating.
The reason I changed my mind is that it seemed like the simplest way to avoid repeating when instantiating a class with 3 levels of inheritance, all emitting the very same warning.